cfme/fixtures/embedded_ansible.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+from wrapanapi.utils.random import random_name
+
+
+def open_namespace(domain_name, namespace_levels):
+    namespace = domain_name
+    for namespace_name in namespace_levels:
+        namespace = namespace.namespaces.instantiate(namespace_name)
+    return namespace
+
+
+def copy_ae_instance(appliance, instance_fqdn, domain_name):
+    """
+    Copy AE Instance into a domain.
```

Comment:
This seems like a complex method to have within the fixtures module, instead of as a method on Instance. 

I see that the `copy_to` method that you're using on the AE Instance, defined here, has a `TODO` for 'Namespace!', which is actually not clear to me what needs to be done there.

https://github.com/ManageIQ/integration_tests/blob/c94723bf2bb7aebd66a1e5e8130fb76f2ff935c2/cfme/automate/explorer/common.py#L30

Can this `Copiable.copy_to` method be expanded to cover your use case here?

Otherwise, can you provide more description of exactly what is going on in this method - there are calls to pop components off of the instance fqdn that aren't clear, and then chained calls to instantiate entities within collections that are not commented and not clear.

Reply:
`Copiable.copy_to` requires instance of an object, but the way that test decoration works, we can't provide AE instance to a fixture. We instead provide a full string path to AE instance, proper classes are then generated inside this function.

I split the call chains into multiple variables so the call is more clear and also provided some explanation and an example.